backend/tracim_backend/lib/core/user.py

```diff
@@ -1121,3 +1123,102 @@ def allowed_to_invite_new_user(self, email: str) -> bool:
             return False
 
         return True
+
+    def create_user_following(
+        self,
+        follower_id: int,
+        leader_id: int,
+        created_date: typing.Optional[datetime.datetime] = None,
+        do_save: bool = True,
+    ) -> UserFollower:
+        # NOTE BS 20201229: sqlalchemy raise database specific error. In addition, error is
+        # raised at commit, so after view execution (so we can't catch it with hapic view
+        # decorators). So check if row exist before try to insert.
+        if (
+            self._session.query(UserFollower)
+            .filter(
+                and_(UserFollower.leader_id == leader_id, UserFollower.follower_id == follower_id)
+            )
+            .count()
+        ):
+            raise UserFollowAlreadyDefined("User follow already defined")
+
+        user_follower = UserFollower(
+            follower_id=follower_id, leader_id=leader_id, created_date=created_date
+        )
+        self._session.add(user_follower)
+
+        if do_save:
+            self._session.flush()
+
+        return user_follower
+
+    def delete_user_following(self, follower_id: int, leader_id: int, do_save: bool = True) -> None:
+        try:
+            user_follow = (
+                self._session.query(UserFollower)
+                .filter(
+                    and_(
+                        UserFollower.leader_id == leader_id, UserFollower.follower_id == follower_id
+                    )
+                )
+                .one()
+            )
+            self._session.delete(user_follow)
+        except NoResultFound:
+            raise NotFound("User following does not exist")
+
+        if do_save:
+            self._session.flush()
```

Comment:
I know this is similar to other methods, but having a this `do_save` parameter (which is always `true`) doesn't seem useful for now (and we can add it easily if needed later).

Reply:
ok, done in next commit